### ISSUE TITLE
Feature: option to export a song list to csv

### ISSUE THREAD

**Issue by MairusuPawa** [UltraStar-Deluxe/USDX#213]
Hello,

Since USDX already scans folders for songs it can use, maybe it could generate a list of those in an relatively easy-to-use format (such as CSV containing song titles / artists) for other purposes? It's possible to extract that data from the db itself, but not straightforward for end-users.

One use case would be checking the available songs directly in a spreadsheet (filtering, ordering, comparing what's supposedly available when a full song pack is installed to what's actually seen on the user's computer in the end to detect missing files, etc); another use case could be printing out song lists easily right before parties.

**Comment by basisbit** [comment]
If I remember correctly, you can already export to csv using the tool [UltraStar Manager](https://sourceforge.net/projects/uman/)

**Comment by bohning** [comment]
Yes, via Extras --> Create Report, you can create song lists in HTML, PDF, TXT and CSV format.

**Comment by RattleSN4K3** [comment]
It would be a good feature, but I doubt it's one with a high priority. _UltraStar Manager_ and equivalent are all providing the same functioning if not better. It's not that it's currently impossible.

_Following paragraphs show a program I used/created, it shouldn't fulfill the request of such exporting feature for UltraStar. Just wanted to mention it. A temporary solution for anyone with the need of an exported list._

Back in the days, I also had the need of a list to print out the songs for parties. I had to create the program as there wasn't really any other with sufficient features. The generated list from my program was an html page which I also copied into document writing software (like Word or Writer [OpenOffice/Libre]) for further formatting. HTML was the main format I used  as you could just upload the file anywhere and anyone could just open the file and see the list digitally (planned; it was the time before the boom of smartphones), recently I also added a filter feature to search for songs. This was in 2008 and I haven't updated the code for the last 7 years. But it still works.
![indexer](https://cloud.githubusercontent.com/assets/6833006/22344998/aa48f56a-e3fe-11e6-9485-e00ee93da68e.png)

It reads the existing txt files, parses the data, checks if the files are valid and list them as a html table with the specified sorting and selected columns (falsely named "row selecting" in the screenshot).

Example list:
downloadable html file (saved/converted with SingleFile extension):
[MySongs_singlefile.htm](https://gist.github.com/RattleSN4K3/8e62a3146f2e00e4e67f3f8e7a93711d/archive/dcb6e84728ed42838bc2525473704a79bed83761.zip)

![list](https://cloud.githubusercontent.com/assets/6833006/22345037/d4ddaca8-e3fe-11e6-820a-065d687d4a3b.png)


If there's a general interest in that tool, let me know - I consider releasing it. As I haven't updated checked the code recently, I would likely have to check for licensing problem (although only open source Javascript libs are used) and if I am really in a mood, I would refactor a lot or even rewrite the program. Not sure how fast I could provide such tool as I am kinda busy.

**Comment by cl9m** [comment]
Hello I search a programm for extract a list from my USDX folder songs, for print it.
@RattleSN4K3 can you share your program same as you say ?
thx

**Comment by pb-programmer** [comment]
> Hello I search a program to extract a list from my USDX folder songs, for print it. @RattleSN4K3 can you share your program same as you say, please ? thx

As RattleSN4K3 said, [UltraStar-Manager](https://github.com/UltraStar-Deluxe/UltraStar-Manager) already offers song list export. It obviously does a lot more stuff as well, but if you just need a quick way to get a song list (as PDF, HTML or CSV), just use UltraStar-Manager, go to "Extras" click on "Create Report" (or use the F8 key) and select what information should be included in the report.

**Comment by cl9m** [comment]
Thank you @pb-programmer I had misread, it's perfect it's exactly what I needed

**Comment by dgruss** [comment]
can be closed?
